### PR TITLE
Editing policy_current_law.json to incorporate ARPA.json values

### PR DESCRIPTION
**Pull request by amshoulders**
This pull request edits the policy_current_law.json file to incorporate the [ARPA.json](https://github.com/PSLmodels/examples/blob/main/psl_examples/taxcalc/ARPA.json) parameter values.

cc @jdebacker

### CODE REVIEW

File: taxcalc/policy_current_law.json [hunk 1]
```diff
@@ -15513,6 +15904,14 @@
...
+            {
+                "year": 2013,
+                "value": 0.0
+            },
+            {
+                "year": 2021,
+                "value": 1000.00
+            },
             {
                 "year": 2013,
```
Comment (jdebacker) [inline taxcalc/policy_current_law.json:15916]: This year should be 2022.  See error message from tests about `CTC_new_c`:  
```
The Value objects for CTC_new_c do not span the specified parameter space. Duplicate combinations:
14085
E           	[((2013,), 2)]
```

**Comment by jdebacker** [comment]
@amshoulders After fixing that year index in the JSON file, you will need to do a couple things to make sure tests pass:

1. You'll need to update the csv files against which output are tested (since these cached files need to match the results produced from the model under current law).  If you run the test suite locally, you'll see a message displayed with the names of the files to replace.  Note that you will only be able to do this with the CPS-based files.  I'll open a PR to your branch to update the PUF-based results.
2. You'll need to update the the `tests/test_reforms.test_round_trip_tcja_reform()` function.  This function starts with 2017 law and layers on tax law changes since to get to current law.  If you look at some in-line comments there you can see how the TCJA and the CARES Act were added.  You'll want to do similar with the ARPA law change.  For the ARPA changes, you can either paste in the dictionary of parameter updates from the  `ARPA.json` file (this is what is done for the CARES Act) OR you can point to the url for  `ARPA.json` file in the PSL/Examples repository.

**Comment by jdebacker** [comment]
I'm don't understand the failure with `test_parameters.py::test_json_file_contents[policy_current_law.json]`.  Here's the error message:

```
>           raise ValueError(failures)
247
E           ValueError: param:<ALD_BusinessLosses_c>; len(value_yrs)=9; known_years={2016, 2017, 2018, 2019, 2026, 2013, 2014, 2015}
248
E           param:<EITC_c>; len(value_yrs)=9; known_years={2016, 2017, 2018, 2019, 2013, 2014, 2015}
249
E           param:<EITC_ps>; len(value_yrs)=9; known_years={2016, 2017, 2018, 2019, 2013, 2014, 2015}
250
E           param:<EITC_InvestIncome_c>; len(value_yrs)=8; known_years={2016, 2017, 2018, 2019, 2013, 2014, 2015}
251
E           param:<CTC_new_ps>; len(value_yrs)=9; known_years={2016, 2017, 2018, 2019, 2013, 2014, 2015}
```

But the `ALD_BusinessLosses_c` looks like it is defined correctly in my view.  I thought gaps in years would be filled in by the extrapolation in ParamTools.  Can anyone let me and @amshoulders know what the issue is here?  Thanks!

cc @hdoupe

**Comment by jdebacker** [comment]
@hdoupe I the issue here with the `ALD_BusinessLosses_c` parameter (highlighted in comment above) an instance where we need to use the `clobber` function you demonstrated in the PSL Demo Day today?

**Comment by hdoupe** [comment]
@jdebacker sorry I just saw this. It looks like this test only loads the JSON file and does not feed it through ParamTools for extrapolation:

https://github.com/PSLmodels/Tax-Calculator/blob/2f94e6e5ab83d1c1c75f210f9a23bafb6bc177d9/taxcalc/tests/test_parameters.py#L204-L252

Maybe a better test would be:
- check the min / max years of the JSON contents
- load JSON with appropriate class (`tc.Policy`, etc.)
- make sure years are filled in correctly

**Comment by jdebacker** [comment]
Ah - thanks @hdoupe! I'll take a stab at rewriting this test.

**Comment by jdebacker** [comment]
@amshoulders This PR looks good to me -- thanks for your work!  You can go ahead and remove the [WIP] from the title.

@MattHJensen I have one question confirming that you are ok with a change to a unit test, but after that, this is ready for your final review.

File: taxcalc/tests/test_parameters.py [hunk 2]
```diff
@@ -232,27 +233,22 @@ def test_json_file_contents(tests_path, fname):
...
-                exp_years = known_years
-            if pname in form_parameters:
-                if defined_years != exp_years:
-                    error = True
-            else:
-                if defined_years != exp_years:
-                    error = True
-            if error:
-                msg = 'param:<{}>; len(value_yrs)={}; known_years={}'
-                fail = msg.format(pname, len(defined_years), exp_years)
```
[the remark this answers is not in the document]
Comment (MattHJensen) [reply taxcalc/tests/test_parameters.py:255]: @jdebacker, do you have a preference here -- keep changes in PR or adopt Hank's suggestion?

File: taxcalc/tests/test_cpscsv.py [hunk 3]
```diff
@@ -63,14 +63,16 @@ def test_agg(tests_path, cps_fullsample):
         raise ValueError(msg)
     # create aggregate diagnostic table using unweighted sub-sample of records
     rn_seed = 180  # to ensure sub-sample is always the same
-    subfrac = 0.03  # sub-sample fraction
+    subfrac = 0.07# 0.03  # sub-sample fraction
```
[the remark this answers is not in the document]
Comment (MattHJensen) [reply taxcalc/tests/test_cpscsv.py:66]: I think your understanding here is correct, and bumping to 7% is fine.

File: taxcalc/tests/test_parameters.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jdebacker) [reply taxcalc/tests/test_parameters.py:255]: I kind of like using the `Consumption/Policy/GrowDiff` objects as they are what is used directly in the model.  @hdoupe do you see this differently?

[the remark this answers is not in the document]
Comment (hdoupe) [reply taxcalc/tests/test_parameters.py:255]: That sounds reasonable to me @jdebacker ! I'm good with the updates to this test. I may suggest some nitpick edits but we can close this thread.

File: taxcalc/tests/test_parameters.py [hunk 4]
```diff
@@ -232,27 +233,22 @@ def test_json_file_contents(tests_path, fname):
...
+    if fname == "consumption.json":
+        o = Consumption()
+    elif fname == "policy_current_law.json":
+        o = Policy()
+    elif fname == "growdiff.json":
+        o = GrowDiff()
+    param_list = []
+    for k, v in o.__dict__.items():
```
Comment (hdoupe) [inline taxcalc/tests/test_parameters.py:243]: @amshoulders Just a nitpick, but objects are iterable so you could do this to make things a little cleaner:

```suggestion
    for k in o: 
```

File: taxcalc/tests/test_parameters.py [hunk 5]
```diff
@@ -232,27 +233,22 @@ def test_json_file_contents(tests_path, fname):
...
+        if k[0].isupper():  # find parameters by case of first letter
+            param_list.append(k)
+    for param in param_list:
+        for y in known_years:
+            o.set_year(y)
+            if np.isnan(o.__getattribute__(param)).any():
```
Comment (hdoupe) [inline taxcalc/tests/test_parameters.py:249]: Another nitpick, but this is equivalent and doesn't require hitting the `__getattribute__` method directly:

```suggestion
            if np.isnan(getattr(o, param)).any():
```

**Comment by MattHJensen** [comment]
Thanks very much @amshoulders for the PR, @jdebacker for your review and contributions, and @hdoupe for your review and suggestions. Merging.

**Pull request merged**